### ISSUE TITLE
Crash while loading mods related to mixin

### ISSUE THREAD

**Issue by StormCircuit** [svenhjol/CharmForge#66]
Read me first!
1. Is your issue about the Fabric version of Charm? Please report your issue here: https://github.com/svenhjol/Charm/issues
2. We don't support Optifine! Please check that the problem still occurs if you remove Optifine from your mods.
3. Don't use the bug report to suggest new features or behavior.

Versions
* Charm version - 2.2.0, 2.2.1, 2.2.2 (all of them crash under similar circumstances)
* Minecraft version - 1.6.5

Description
Minecraft crashes on load with forge 1.16.5 - 36.0.22

Optional: Steps To Reproduce
Using just Quark-r2.4-304, AutoRegLib-1.6-47, and charm-forge-1.16.4-2.2.2 Minecraft crashes while loading mods. No crash report is generated but through examining 'latest.log' the error appears to be related to mixin. Pastebin link below.

When downgrading to charm-forge-1.16.4-2.2.1 the crashes disappear. But with a larger set of mods I crash with all versions of charm for 1.16.4 (2.2.2, 2.2.1, 2.2.0)

Optional: Stacktrace/Screenshots
latest.log log for large set of mods:
https://pastebin.com/RTDCdz8g

screenshot of mods that always crash with Charm:
![image](https://user-images.githubusercontent.com/16582422/107824525-7a277580-6d36-11eb-958e-d072b614c45e.png)

latest.log for just quark, autoreglib, and charm 2.2.2:
https://pastebin.com/UeeSRSnn

**Comment by StormCircuit** [comment]
I apologize if the pseudo 2-in-1 bug report or bug reporting with the larger set of mods is unconventional or not proper. Please let me know if I can correct the bug report.

**Comment by WenXin20** [comment]
I also got this crash, but only after updating to forge v36.0.22, v36.0.7 did not crash with it.
https://gist.github.com/WenXin20/397719fdcfcdcab0959e40648920f425

**Comment by StormCircuit** [comment]
I can confirm WenXin20's solution, actually I downgraded to v35.0.21 and it stopped crashing.

**Comment by GhostRiderGrey** [comment]
Same issue here using forge v36.0.22.  Downgrading to v36.0.21 fixes the crashing issue.

**Comment by Zothen1973** [comment]
Downgrading is NO fix! 

Server still crashes with this mod in Forge 36.0.42.

**Comment by Hubry** [comment]
Conflicts with a change made in Forge 36.0.22: https://github.com/MinecraftForge/MinecraftForge/blob/1.16.x/patches/minecraft/net/minecraft/util/Util.java.patch, which makes the logging quieter.

**Comment by TelepathicGrunt** [comment]
Yep. What Hubry said. As of Forge 36.0.22, the datafixer mixin is no longer needed. Speaking of which, I should remove my datafixer workaround from my mod as well lol

**Comment by Elspar2** [comment]
This is an issue for me too.

**Comment by Zothen1973** [comment]
Newest version still crashes on server side 1.16.5 Forge 36.0.42.

**Comment by svenhjol** [comment]
> Newest version still crashes on server side 1.16.5 Forge 36.0.42.

Please post your crashlog

**Comment by Zothen1973** [comment]
I would, but no crashlog is generated, only the standard logs.

[latest.log](https://github.com/svenhjol/CharmForge/files/6036692/latest.log)

**Comment by Hubry** [comment]
Doesn't look like a fault of Charm, but rather the conflict between Quark and *literally every single MCreator mod with worldgen*. MCreator generated code's quality is so shoddy...

Anyways, I would recommend putting a `require = 0` on this injection so that if anything changes again it doesn't crash - this feature is so insequential that crashing from this is many orders of magnitude more annoying than seeing these few lines of warnings.

**Comment by svenhjol** [comment]
The mixin responsible for the error (`UtilMixin`) was removed in the latest version of CharmForge (2.3.2). If there's another mixin causing the issue, please let me know which one to sort.

**Comment by BugmanBugman** [comment]
> Doesn't look like a fault of Charm, but rather the conflict between Quark and _literally every single MCreator mod with worldgen_. MCreator generated code's quality is so shoddy...
> 
> Anyways, I would recommend putting a `require = 0` on this injection so that if anything changes again it doesn't crash - this feature is so inconsequential, crashing from this is many orders of magnitude more annoying than seeing these few lines of warnings.

I’m getting a launch crash but I don’t have any mcreator mods

**Comment by TelepathicGrunt** [comment]
Can you try sharing the latest.log file? It may have more info. You can get the log file by going into the logs folder above the mods folder and find latest.log inside it

**Comment by svenhjol** [comment]
@BugmanBugman is that happening with CharmForge 2.3.2 or 2.2.2? If the latter, please update and try again.

**Comment by Zothen1973** [comment]
CharmForge 2.3.2

I attached the latest.log in my last comment.

**Comment by svenhjol** [comment]
@Zothen1973 don't worry, that's something that would be done by a developer in their mod.

Looking at your latest.log, it's crashing out with 
```
[24Feb2021 15:36:23.787] [main/FATAL] [net.minecraft.server.Main/]: Failed to start the minecraft server
java.lang.NullPointerException: null
[...]
at vazkii.quark.base.world.WorldGenHandler.onBiomesLoaded(WorldGenHandler.java:161) ~[quark:r2.4-306]
```

@Hubry I guess the above is what you were saying about worldgen crashes with MCreator mods?

**Comment by TelepathicGrunt** [comment]
Yeah. Quark crashes because mcreator makes ConfiguredFeatures that are not registered and sometimes cannot be parsed to json. Mcreator really should fix that as it’s concerning to have broken ConfiguredFeatures lol

**Comment by Zothen1973** [comment]
Without "Charm", my server + Quark starts without any crash. Even with worldgen mods like BoP and/or Byg,

Other mods like "Repurposed Structures" are causing a crash too, though.

**Comment by TelepathicGrunt** [comment]
@Zothen1973 Come over to my github then if my mod repurposed structures is crashing. My mod too will crash due to broken configuredfeatures from other mods. I can help you find which feature and mod it is coming from. Basically, remove all crashing mods and then put on the mod Blame. Start the world. And then give me the latest.log for me to read Blame's output

**Comment by Zothen1973** [comment]
Will do later this evening. I am still "at work" aka home office ;)

**Comment by BugmanBugman** [comment]
Deleting the config seemed to have to fixed it for me

**Comment by StormCircuit** [comment]
> 
> Deleting the config seemed to have to fixed it for me

as in deleting the config for charm and using the latest version of both charm and forge fixed it?

**Comment by Staegrin** [comment]
Deleting the old config does fix the crash at startup.

**Comment by StormCircuit** [comment]
I can also confirm deleting the config has fixed it for me.

**Comment by StormCircuit** [comment]
Updating again to say that you need to delete the config every time you start or it will crash over and over.
